lib/assert.js

```diff
@@ -690,6 +691,15 @@ class Assertions {
 		});
 
 		this.snapshot = withSkip((expected, ...rest) => {
+			if (disableSnapshots && experiments.disableSnapshotsInHooks) {
+				fail(new AssertionError({
+					assertion: 'snapshot',
+					message: '`t.snapshot()` cannot be used here',
```

Comment:
If I understand it correctly, `t.snapshot()` and `t.try` are only allowed to be used in tests.
If that's so, I would suggest changing error to something like
```
`t.snapshot()` can be used in tests only
```

Reply:
Yes. Changing to "can only be used in tests"